Add buttons to search alerts & LC archive near source on source page

acrellin opened the pull request:
This PR introduces a button to the source page that, when clicked, submits a request to query alerts within 3 arcseconds of the source in question and redirects to the alerts page to render the results. Other improvements to the alerts page, including a spinner during request processing, and redux state structure updates, are included.

Closes  https://github.com/fritz-marshal/fritz/issues/287

![alerts_search_button](https://user-images.githubusercontent.com/7230285/120868269-87b47580-c548-11eb-8c45-939efedd6194.gif)

acrellin commented:
CI failure unrelated

kmshin1397 commented on extensions/skyportal/static/js/components/ArchiveSearchButton.jsx:
```diff
@@ -0,0 +1,61 @@
+import { showNotification } from "baselayer/components/Notifications";
+import * as archiveActions from "../ducks/archive";
+
+const AlertsSearchButton = ({ ra, dec, radius = 3 }) => {
```
```suggestion
const ArchiveSearchButton = ({ ra, dec, radius = 3 }) => {
```

kmshin1397 commented on extensions/skyportal/static/js/components/ArchiveSearchButton.jsx:
```diff
@@ -0,0 +1,61 @@
+  );
+};
+
+AlertsSearchButton.propTypes = {
```
```suggestion
ArchiveSearchButton.propTypes = {
```

kmshin1397 commented on extensions/skyportal/static/js/components/ArchiveSearchButton.jsx:
```diff
@@ -0,0 +1,61 @@
+  dec: PropTypes.number.isRequired,
+  radius: PropTypes.number,
+};
+AlertsSearchButton.defaultProps = {
```
```suggestion
ArchiveSearchButton.defaultProps = {
```

kmshin1397 commented on extensions/skyportal/static/js/components/ArchiveSearchButton.jsx:
```diff
@@ -0,0 +1,61 @@
+  radius: 3,
+};
+
+export default AlertsSearchButton;
```
```suggestion
export default ArchiveSearchButton;
```

kmshin1397 reviewed (changes requested):
Looks good other than some renaming, thanks @acrellin!

acrellin commented:
@kmshin1397 ah right, of course. Fixed!

dmitryduev commented on extensions/skyportal/static/js/components/ArchiveSearchButton.jsx:
```diff
@@ -0,0 +1,61 @@
+
+  return (
+    <Link to="/archive" onClick={handleClick}>
+      <Button variant="contained">View ZTF Light Curve Archive</Button>
```
```suggestion
      <Button variant="contained">Search ZTF Light Curve Archive</Button>
```

dmitryduev commented on extensions/skyportal/static/js/components/AlertsSearchButton.jsx:
```diff
@@ -0,0 +1,34 @@
+  return (
+    <Link to="/alerts" onClick={handleClick}>
+      <Button variant="contained">
+        Search Nearby Alerts
```
```suggestion
        Search ZTF Alert Archive
```

dmitryduev commented on extensions/skyportal/static/js/components/AlertsSearchButton.jsx:
```diff
@@ -0,0 +1,34 @@
+const AlertsSearchButton = ({ objID, ra, dec, radius = 3 }) => {
+  const dispatch = useDispatch();
+  const handleClick = () => {
+    dispatch(alertsActions.fetchAlerts({ object_id: objID, ra, dec, radius }));
```
There may be sources with other `objectId`'s within 3 arcseconds - let's not restrict the search to the current `objID` only:

```suggestion
    dispatch(alertsActions.fetchAlerts({ ra, dec, radius }));
```

dmitryduev commented on extensions/skyportal/static/js/components/SourceDesktop.jsx:
```diff
@@ -244,6 +246,16 @@ const SourceDesktop = ({ source }) => {
           </div>
+          <div className={classes.infoLine}>
+            <div className={classes.sourceInfo}>
+              <AlertsSearchButton objID={source.id} ra={source.ra} dec={source.dec} />
```
```suggestion
              <AlertsSearchButton ra={source.ra} dec={source.dec} />
```

dmitryduev commented on extensions/skyportal/static/js/components/AlertsSearchButton.jsx:
```diff
@@ -0,0 +1,34 @@
+
+import * as alertsActions from "../ducks/alerts";
+
+const AlertsSearchButton = ({ objID, ra, dec, radius = 3 }) => {
```
```suggestion
const AlertsSearchButton = ({ ra, dec, radius = 3 }) => {
```

dmitryduev commented on extensions/skyportal/static/js/components/AlertsSearchButton.jsx:
```diff
@@ -0,0 +1,34 @@
+};
+
+AlertsSearchButton.propTypes = {
+  objID: PropTypes.string.isRequired,
```
```suggestion
```

dmitryduev reviewed (changes requested):
Looks great, thanks @acrellin! A few requests:

Could you please: 
- Make the buttons the same size (i.e. smaller) as the ones below? 
- Move them up so that they appear right after the coordinates (since we'll be querying by coordinates)?
- Unrelated to this PR, but I'd swap redshift and TNS match info in the header - the latter is also done by coordinates, so would be more natural to appear after the buttons from this PR.
- How hard would it be to fill out the forms on the Alerts/Archive pages with the ra/dec/radius info so that the user can see what was searched?

acrellin commented:
@dmitryduev I've addressed all of your comments except for the form population which would require storing the form-values-to-be in redux state and populating the component's inputs accordingly. Doesn't seem like there's much added benefit for that work (they can easily see the coords of the source from which they clicked search). Thoughts?

dmitryduev reviewed (approved):
👍🏻 
re: autofill - let's leave it for now.
